main.go

```diff
@@ -57,6 +75,37 @@ func main() {
 		prometheus.MustRegister(dhis2Exporter)
 	}
 
+	// Register SendGrid exporters
+	apiKeys := make(map[string]string)
+	for _, account := range config.SendGridAccounts {
+		apiKeys[account.AccountName] = account.APIKey
+	}
+	sendgridExporter := sendgrid.NewExporter(apiKeys)
+	prometheus.MustRegister(sendgridExporter)
+
 	http.Handle("/metrics", promhttp.Handler())
-	http.ListenAndServe(":8080", nil)
+	log.Printf("Starting server on %s", *listenAddress)
+
+	httpServer := http.Server{
+		Addr: *listenAddress,
+	}
+
+	idleConnectionsClosed := make(chan struct{})
```

Comment:
Here's your revised text with improved grammar and clarity:

1. It would be better to move the graceful shutdown to a method, or at least add a comment explaining its purpose.
2. I suggest using a context with a timeout to ensure the server doesn’t hang indefinitely during shutdown.

If these tasks are time-consuming, feel free to add a "TODO:" comment for now

Reply:
@roypeter i had made the following changes as per your request.
1. Moved Graceful Shutdown to a Method:

I refactored the graceful shutdown logic into a separate gracefulShutdown method to improve code organization and readability. This makes the main function cleaner and separates shutdown logic from startup code. Added comments in the gracefulShutdown function to explain its purpose and how it works.

2. Added Context with Timeout:

I introduced a context with a 5-second timeout for the server shutdown to prevent indefinite hangs. This ensures that the server will attempt to shut down gracefully within a defined period, improving reliability and control over the shutdown process.

Reply:
It looks a bit complicated after refactoring; perhaps we should just keep it in main.go